src/messages.rs

```diff
@@ -117,7 +100,7 @@ pub enum Request {
 
 #[derive(PartialEq, Eq, Clone, Debug, RustcEncodable, RustcDecodable)]
 pub enum Response {
-    Get(GetDataResponse),
+    Get(Data),
```

Comment:
For External responses we attach the original request inside the message `Response` enum; eg ErrorReturn has it - this is different from InternalResponse that has it explicitly in the messagetype enum - it's something that needs ironing out.  With this change Response::Get(data) would not have an original request anymore though.